providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/SyncMetadataHook.java

```diff
@@ -0,0 +1,26 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.Hook;
+import dev.openfeature.sdk.HookContext;
+
+class SyncMetadataHook implements Hook<Object> {
+
+    private Supplier<EvaluationContext> contextSupplier;
+
+    SyncMetadataHook(Supplier<EvaluationContext> contextSupplier) {
+        this.contextSupplier = contextSupplier;
+    }
+
+    /**
+     * Return the enriched context, including the additional attributes from the sync-metadata.
```

Comment:
I am not sure this doc is correct; we're not enriching context; we're only returning the result of the supplier - should we merge this with the provided ctx?

Reply:
No, we shouldn't merge anything, because the SDK will merge this result into things before the provider evaluates; that was the weird thing about the previous implementation; the SDK should do all the merging.

The comment is wrong though, as you point out. I will update it.